test/unit/prbuf.c

```diff
@@ -328,6 +328,65 @@ test_buffer_prepared_large(void)
 	footer();
 }
 
+/**
+ * Check:
+ * - we can alloc record of prbuf_max_record_size
+ * - we can't alloc larger buffer
+ * - record of max size is actually usable
+ */
+static void
+test_max_record_size(void)
+{
+	header();
+
+	size_t i;
+	for (i = 0; i < lengthof(buffer_size_arr); i++) {
+		uint32_t buffer_size = buffer_size_arr[i];
+		struct prbuf buf;
+		char *mem = xmalloc(buffer_size);
+
+		prbuf_create(&buf, mem, buffer_size);
+		size_t max_size = prbuf_max_record_size(&buf);
+
+		size_t j;
+		char *payload = xmalloc(max_size);
```

Comment:
Changed test a bit:
- drop testing for different buffer sizes as code of interest does not depend on the size
- use static buffers (as in other tests)
- changed unit tests messages so that we can have some information of what's going on if test fails (reporting
  actual values). I'm actually changed the other tests too in future PR where I'm switching to new prbuf reader API.